templates/collection.nomad

```diff
@@ -92,9 +92,39 @@ job "collection-${name}" {
           pg = 5432
         }
       }
-      env {
-        POSTGRES_USER = "snoop"
-        POSTGRES_DATABASE = "snoop"
+      template {
+        data = <<EOF
+          POSTGRES_USER = "snoop"
+          POSTGRES_DATABASE = "snoop"
+          {{- with secret "liquid/collections/${name}/snoop.postgres" }}
+            POSTGRES_PASSWORD = {{.Data.secret_key}}
+          {{- end }}
+        EOF
+        destination = "local/postgres.env"
+        env = true
+      }
+      template {
+        data = <<-EOF
+        #!/bin/sh
+        set -ex
+        pwd
+        date
+        if grep -Fq "$host all all all trust" $PGDATA/pg_hba.conf
+        then
+          psql -U snoop -c "ALTER USER snoop password '$POSTGRES_PASSWORD'"
+          sed -i '$d' $PGDATA/pg_hba.conf
+          sed -i '$d' $PGDATA/pg_hba.conf
+          {
+            echo
+            echo "host all all all md5"
```

Comment:
Do we have any other option than `md5`? Maybe `sha1`? Because MD5 has been shown to be weak for a while now - https://en.wikipedia.org/wiki/MD5#Security

Reply:
It's the default of postgres, and what the container will do in its entrypoint if the password is set during the first startup, that's why I used it. I'll check whether we can change that.

Reply:
I'll just leave this here:
https://stackoverflow.com/questions/43333053/change-postgresql-password-encryption-from-md5-to-sha

Reply:
I would bet **zero** hassle, but we need to check that all of our Hoover tables will survive the upgrade. Can you check this out on your local machine? Just `deploy`, update all `postgresql` docker tags, `deploy` again, and see if anything is broken when browsing Hoover files.

If we do this migration now as a dependency for this pull request, please open a different pull request for upgrading everything to pg 10.

Reply:
It is a little more hassle actually. The data needs to be dumped and then reimported. Postgres 10 throws an error when you try to start a 10.x container with a 9.6 db: 
` The data directory was initialized by PostgreSQL version 9.6, which is not compatible with this version 10.9 (Debian 10.9-1.pgdg90+1).`

Reply:
Yeah, the upgrade process for postgres is rather involved. They have a [`pg_upgrade`](https://www.postgresql.org/docs/10/pgupgrade.html) command that needs the old postgres binary, the new one, the old data dir, and a place for the new data dir.